### PR TITLE
add Rest tests to index-lifecycle

### PR DESCRIPTION
**Pull request by talevy**
This PR introduces a `qa` module within the index-lifecycle project.

the idea is to have both complex policies tested, as well as
policies with isolated/singular actions. So far, only tests with policies
containing one action are implemented.

Following Actions have implemented tests in this commit

- AllocateAction
- DeleteAction
- ForceMergeAction
- ReadOnlyAction
- ReplicasAction

tests to be added later

- RolloverAction
- ShrinkAction

### CODE REVIEW

File: x-pack/plugin/index-lifecycle/qa/src/test/java/org/elasticsearch/xpack/indexlifecycle/TimeSeriesLifecycleActionsIT.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
+/*
```
Comment (talevy) [inline x-pack/plugin/index-lifecycle/qa/src/test/java/org/elasticsearch/xpack/indexlifecycle/TimeSeriesLifecycleActionsIT.java]: woops. wrong license. will fix

**Comment by talevy** [comment]
after discussing how to handle `qa` tests with the team, we came to the conclusion that these tests should be a part of the existing `index-lifecycle` project, not a sub-project. I will refactor here

**Comment by talevy** [comment]
structure has been updated. thanks to some changes made in `master`, I've re-enabled `integTest`, modified the integTestCluster to launch two nodes, and added some simple isolated rest tests.

@colings86 do you mind taking another look?

**Review by colings86 — COMMENTED** [review]
Mostly LGTM but I noticed we are missing a test for rollover here. Is that intentional?

**Pull request merged**